src/annotator/sidebar.coffee

```diff
@@ -172,42 +176,43 @@ module.exports = class Sidebar extends Host
     # The sidebar iframe is hidden or shown by adjusting the left margin of its
     # container.
 
-    if @onLayoutChange
-      frame = @frame || @externalFrame
-      rect = frame[0].getBoundingClientRect()
-      computedStyle = window.getComputedStyle(frame[0])
-      width = parseInt(computedStyle.width)
-      leftMargin = parseInt(computedStyle.marginLeft)
-
-      # The width of the sidebar that is visible on screen, including the
-      # toolbar, which is always visible.
-      frameVisibleWidth = toolbarWidth
-
-      if explicitExpandedState?
-        # When we are explicitly saying to open or close, jump
-        # straight to the upper and lower bounding widths.
-        if explicitExpandedState
-          frameVisibleWidth += width
+    frame = @frame || @externalFrame
```

Comment:
The meat of this approach is that it takes advantage of layout-change logic that already exists...